packages/ember-runtime/lib/computed/reduce_computed_macros.js

```diff
@@ -362,6 +364,49 @@ export function uniq(...args) {
   });
 }
 
+if (isEnabled('ember-runtime-computed-uniq-by')) {
+/**
+  A computed property which returns a new array with all the unique
+  elements from an array, with uniqueness determined by specific key.
+  Example
+  ```javascript
+  var Hamster = Ember.Object.extend({
+    uniqueFruits: Ember.computed.uniqBy('fruits', 'id')
+  });
+  var hamster = Hamster.create({
+    fruits: [
+      { id: 1, 'banana' },
+      { id: 2, 'grape' },
+      { id: 3, 'peach' },
+      { id: 1, 'banana' }
+    ]
+  });
+  hamster.get('uniqueFruits'); // [ { id: 1, 'banana' }, { id: 2, 'grape' }, { id: 3, 'peach' }]
+  ```
+  @method uniqBy
+  @for Ember.computed
+  @param {String} dependentKey
+  @param {String} propertyKey
+  @return {Ember.ComputedProperty} computes a new array with all the
+  unique elements from the dependent array
+  @public
+*/
+export function uniqBy(dependentKey, propertyKey) {
```

Comment:
This fails JSHint b/c the function is declared inside the `if` block. Please define the function outside then export inside the `if`, or if that still complains the flagging in `packages/ember-runtime/lib/index.js` would be sufficient.